command/rootchain/emit/emit.go

```diff
@@ -76,7 +86,14 @@ func runCommand(cmd *cobra.Command, _ []string) {
 		return
 	}
 
-	pendingNonce, err := helper.GetPendingNonce(helper.GetRootchainAdminAddr())
+	txRelayer, err := txrelayer.NewTxRelayer(jsonRPCAddress)
+	if err != nil {
+		outputter.SetError(fmt.Errorf("could not create rootchain interactor: %w", err))
+
+		return
+	}
+
+	pendingNonce, err := txRelayer.GetNonce(ethgo.Address(helper.GetRootchainAdminAddr()))
```

Comment:
The `relayer` handles the nonce already.

Reply:
Yes it does, but it doesn't cover all the use cases.
emit command is submitting transactions in parallel, which means that nonce must be set from the outside in order to be 
valid. https://github.com/0xPolygon/polygon-edge/blob/feature/v3-parity/command/rootchain/emit/emit.go#L93.
Therefore I thought this is the proper way to handle it, expose `GetNonce` and use it to set nonce when creating transaction instances. Do you have any more elegant approach?

Reply:
Lock the `SendTransaction` call. That way the nonce will be correct. It is a naive approach and it does not work for all use cases but it works for this.

Reply:
Been there done that. If I remember correctly bridge e2e test started to fail, since rootchain emit command runs sequentially in that case. :disappointed: 
So there is no point we made emit command sending transactions in parallel (which was done in order to make bridge e2e test works faster).

Reply:
Ok, so it is not that it does not work, but that it is not as fast for e2e which is another thing. Ok, I remember now, you have to lock only between getting the nonce and sending the transaction, unlock for receipt wait.

Once you send the transaction, the internal txpool gets updated and the next query nonce is sequential. This should fix both things.

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/989/commits/f2a1c83e5f8631e495529caa64ce2e381553e249